bot/exts/fun/off_topic_names.py

```diff
@@ -51,24 +60,129 @@ async def update_names(self) -> None:
+        successfully_renamed = set()
+        for attempt in range(MAX_RENAME_ATTEMPTS):
+            # Get the necessary amount of new channel names
+            remaining_channels = set(channel.id for channel in channels if channel.id not in successfully_renamed)
+            try:
+                num_names_to_fetch = len(remaining_channels)
+                new_channel_names: list = await self.bot.api_client.get(
+                    "bot/off-topic-channel-names", params={"random_items": num_names_to_fetch}
+                )
+                if (num_fetched := len(new_channel_names)) < num_names_to_fetch:
+                    # Handle when there's not enough names in the pool
+                    log.warn(
+                        f"Attempted to fetch {num_names_to_fetch} off-topic names but "
+                        f"only received {num_fetched} so not updating any off-topic names."
+                    )
+                    if failed_renames:
+                        # Make sure we still notify of any previous rename failures
+                        await self.handle_failed_renames(failed_renames, channels)
+                    return
+                new_channel_names = iter(new_channel_names)
+            except ResponseCodeError as e:
+                log.error(f"Failed to get new off-topic channel names: code {e.response.status}")
+                raise
+
+            # Attempt to rename remaining channels using the acquired names.
+            # NB: We loop of `channels` instead of `remaining_channels` to ensure
+            # that the `channel_indx` is the same as the `number` to pass into `OTN_FORMATTER`
+            for channel_indx, channel in enumerate(channels):
+                if channel.id not in remaining_channels:
+                    continue
+
+                new_channel_name = next(new_channel_names)
+
+                name_before = channel.name
+                name_after = OTN_FORMATTER.format(number=channel_indx, name=new_channel_name)
+                log.debug(
+                    f"Attempt #{attempt + 1} / {MAX_RENAME_ATTEMPTS} is attempting to rename "
+                    f"#{name_before} to #{name_after}"
+                )
+                try:
+                    await channel.edit(name=name_after)
+                    successfully_renamed.add(channel.id)
+                    log.debug(f"Updated off-topic name #{name_before} to #{name_after}")
+                except HTTPException as e:
+                    # We need to handle code 50035 ("invalid form body"),
+                    # which we get when the new channel name isn't allowed.
+                    #
+                    # For more information see https://github.com/python-discord/bot/issues/2500
+                    if (e.code != 50035):
+                        # The error isn't the one we want to handle so re-raise
+                        raise e
+
+                    failed_renames[channel_indx] += [new_channel_name]
```

Comment:
You can directly use append here
```suggestion
                    failed_renames[channel_indx].append(new_channel_name)
```

Reply:
Good point, done in bf9f025